lastminutewf2020.yml

```
@@ -36,6 +65,23 @@
         mode: '0644'
 - hosts: all
   tasks:
+    # TODO boo, this task always runs
+    - name: update root password (if root_password already crypted)
+      ansible.builtin.user:
+        name: "root"
+        state: present
+        password: "{{ root_password }}"
+      when:
+        - root_password is defined
+        - '"$6$" in root_password'
+    - name: update root password (if root_password is cleartext)
+      ansible.builtin.user:
+        name: "root"
+        state: present
+        password: "{{ root_password | password_hash('sha512') }}"
```

This becomes more idempotent when doing this:
```suggestion
        password: "{{ root_password | password_hash('sha512', root_salt ) }}"
```
the root_salt should be set per contest (and semi-hard to guess)

---

interesting put the salt here causes it to create a stable crypt, that it doesn't try to replace every thing.   Unlike on my pre-crypted version where it does each time.